utils/nonlocaluserkey_functions.go

```diff
@@ -0,0 +1,179 @@
+package utils
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	api_latest "github.com/vast-data/terraform-provider-vastdata/codegen/latest"
+	vast_client "github.com/vast-data/terraform-provider-vastdata/vast-client"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+func CreateNonLocalUserKeyFunc(ctx context.Context, _client any, attr map[string]any, data map[string]any, headers map[string]string) (*http.Response, error) {
+	client := _client.(*vast_client.VMSSession)
+	attributes, err := getAttributesAsString([]string{"path"}, attr)
+	if err != nil {
+		return nil, err
+	}
+	path := (*attributes)["path"]
+	uid := data["uid"]
+	tenantId := data["tenant_id"]
+	enabled, ok := data["enabled"]
+	if !ok {
+		enabled = false
+	}
+	buffer, marshallingError := json.Marshal(data)
+	if marshallingError != nil {
+		return nil, marshallingError
+	}
+	tflog.Debug(ctx, fmt.Sprintf("Calling POST to path \"%v\"", attr))
+	response, err := client.Post(ctx, path, bytes.NewReader(buffer), map[string]string{})
+	if err != nil {
+		return response, err
+	}
+	responseBody := map[string]any{}
+	err = UnmarshalBodyToMap(response, &responseBody)
+	if err != nil {
+		return nil, err
+	}
+	accessKey := responseBody["access_key"]
+	responseBody["id"] = fmt.Sprintf("%v-%v", uid, accessKey)
+	responseBody["uid"] = uid
+	responseBody["enabled"] = enabled
+	responseBody["tenant_id"] = tenantId
+	if !(enabled.(bool)) {
+		payload := map[string]any{"uid": uid, "access_key": accessKey, "enabled": false, "tenant_id": tenantId}
+		pBuffer, pMarshallingError := json.Marshal(payload)
+		if pMarshallingError != nil {
+			return nil, pMarshallingError
+		}
+		tflog.Debug(ctx, fmt.Sprintf("Disable NonLocalUserKey %v", accessKey))
```

Comment:
```suggestion
		tflog.Debug(ctx, fmt.Sprintf("Disabling NonLocalUserKey %v", accessKey))
```

I supposed there's no way to do it in one request?